### PR TITLE
[XM] Update bindings and tests for Xcode8b3

### PR DESCRIPTION
**Pull request by chamons**
- New AppKit bindings
- Fixes for bindings to get introspection tests passing on 10.12

### CODE REVIEW

**Comment by chamons** [comment]
@dalexsoto - Please review the intents changes. 
Anyone - please review everything else :)

File: src/appkit.cs [hunk 1]
```diff
@@ -2125,6 +2125,10 @@ public interface NSButton : NSAccessibilityButton {
 		[Mac (10,12)]
 		[Export ("imageHugsTitle")]
 		bool ImageHugsTitle { get; set; }
+
+		[Mac (10,5)]
```
Comment (spouliot) [inline src/appkit.cs]: trivial but we generally don't add availability attributes for versions lower than our minimum support (e.g. 10.7)

File: src/appkit.cs [hunk 2]
```diff
@@ -5245,6 +5249,15 @@ public partial interface NSDocument {
...
 #endif
+
+		[Mac (10,12)]
+		[Export ("isBrowsingVersions")] 
+		bool IsBrowsingVersions { get; }
+
+
+		[Mac (10,12)]
+		[Export ("stopBrowsingVersionsWithCompletionHandler:")]
+		void StopBrowsingVersions (Action completionHandler);
```
Comment (spouliot) [inline src/appkit.cs]: `[Async]` ?

File: src/appkit.cs (hunk 1 above)
Comment (chamons) [inline src/appkit.cs]: Yeah. It was mostly there for "this is new in the header but with an old support number, which was strange enough to note"

**Comment by xamarin-release-manager** [comment]
Build failure

File: src/appkit.cs (hunk 1 above)
Comment (spouliot) [inline src/appkit.cs]: yeah, those are always worth checking (wrench bots will do it) as they are often mistakes

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by dalexsoto** [comment]
Intents changes LGTM 👍 and everything else, thanks for fixing that part :)

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by spouliot** [comment]
unrelated, random, crash

**Pull request merged**